tests/unit/capi/GEOSGeom_transformXYZTest.cpp

```diff
@@ -0,0 +1,333 @@
+//
+// Test Suite for C-API GEOSGeom_transformXYZ
+
+#include <tut/tut.hpp>
+// geos
+#include <geos_c.h>
+// std
+#include <cstdarg>
+#include <cstdio>
+#include <cstdlib>
+
+#include "capi_test_utils.h"
+
+namespace tut {
+struct test_capi_geosgeom_transformxyz : public capitest::utility {};
+
+typedef test_group<test_capi_geosgeom_transformxyz> group;
+typedef group::object object;
+
+group test_capi_geosgeom_transformxyz_group("capi::GEOSGeom_transformXYZ");
+
+static int SCALE_2_3_4(double* x, double* y, double* z, void* userdata) {
+    (*x) *= 2;
+    (*y) *= 3;
+    (*z) *= 4;
+    (void)(userdata);  // make unused parameter warning go away
+    return 1;
+}
+
+// callback that doesn't update coordinates should return original values
+template <>
+template <>
+void object::test<1>() {
+    GEOSGeometry* geom = GEOSGeomFromWKT("POINT (1 1 1)");
+
+    GEOSGeometry* out = GEOSGeom_transformXYZ(
+        geom,
+        [](double* x, double* y, double* z, void* userdata) {
+            (void)(x);         // make unused parameter warning go away
+            (void)(y);         // make unused parameter warning go away
+            (void)(z);         // make unused parameter warning go away
+            (void)(userdata);  // make unused parameter warning go away
+            return 1;
+        },
+        nullptr);
+
+    ensure(out != nullptr);
+    ensure_equals(GEOSEqualsExact(out, geom, 0), 1);
+
+    GEOSGeom_destroy(geom);
+    GEOSGeom_destroy(out);
+}
+
+// failed callback should return NULL
+template <>
+template <>
+void object::test<2>() {
+    GEOSGeometry* geom = GEOSGeomFromWKT("POINT (1 1 1)");
+
+    GEOSGeometry* out = GEOSGeom_transformXYZ(
+        geom,
+        [](double* x, double* y, double* z, void* userdata) {
+            (void)(x);         // make unused parameter warning go away
+            (void)(y);         // make unused parameter warning go away
+            (void)(z);         // make unused parameter warning go away
+            (void)(userdata);  // make unused parameter warning go away
+            return 0;          // indicates error
+        },
+        nullptr);
+
+    ensure(out == nullptr);
+
+    GEOSGeom_destroy(geom);
+}
+
+// callback should modify point coordinates
+template <>
+template <>
+void object::test<3>() {
+    GEOSGeometry* geom = GEOSGeomFromWKT("POINT (1 1 1)");
+
+    GEOSGeometry* out = GEOSGeom_transformXYZ(geom, SCALE_2_3_4, nullptr);
+
+    ensure(out != nullptr);
+    ensure_geometry_equals(out, "POINT (2 3 4)");
+
+    double xmin, ymin, xmax, ymax;
+    GEOSGeom_getExtent(out, &xmin, &ymin, &xmax, &ymax);
+    ensure_equals(xmin, 2);
+    ensure_equals(ymin, 3);
+    ensure_equals(xmax, 2);
+    ensure_equals(ymax, 3);
+
```

Comment:
Test Z result?

Reply:
Also apply to the next comment: the geometry is tested with `ensure_geometry_equals` on the lines above, so maybe we can just remove the extent test ? Alternatively, what other check would be appropriate ?